docs/software/spack.md

```diff
@@ -0,0 +1,157 @@
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack install --add gcc@13.1.0
+```
+
+Once the installation is complete, you can make it available to Spack with `spack compiler add`:
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack compiler add <path to compiler>
+```
+
+>Note that, in this case, you can locate the path of the compiler with `spack location -i gcc@13.1.0`.
+
+Once the compiler is added, you can install any subsequent packages using the compiler you've installed with the `%` operator. For example:
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack install --add fastqc%gcc@13.1.0
+```
+
+### Key Spack Terms: 
+
+| Term | Definition |
+|---------|----------|
+| Spec | Description of the build requirements for an installation. This includes the software, version, compiler, and other variants. |
+| Variant | An option or feature that can be toggled for a specific package. These usually correspond to a configuration or compilation setting. |
+| Repo | A central location which stores packages. Repo locations can be configured. |
+| Environment | A collection of packages and dependencies that are available to a user. In Spack, environments do not interact. |
+
+### Basic Spack commands:
+
+| Command | Function |
+|---------|----------|
+| `spack list` | Find and search for available packages by name |
+| `spack info` | Display detailed information about a package, including versions and dependencies |
+| `spack env create <environment name>` | Create a Spack environment |
+| `spacktivate <environment name>` | Activates a Spack environment |
+| `despacktivate` | Deactivates a Spack environment |
+| `spack find` | List and search packages installed in the active environment |
+| `spack install --add <software>` | Adds specs for selected package and installs it in the active environment |
+| `spack compilers` | Lists available compilers |
+| `spack spec <software>` | Shows all dependencies and items that would be installed if `<software>` is installed > |
+
+### Troubleshooting Spack
+
+If an installation fails, there are several ways to troubleshoot the failure. Common issues with installations include: 
+
+* __Building with the wrong compiler:__ Double check if the compiler you are using is compatible with the software you are attempting to build. 
+* __Building with an incompatible variant set:__ Confirm that the configuration settings for the software are compatible and as expected. If not, try enabling or disabling variants that may be causing problems. 
+* __Building an unexpected version:__ Make sure you are building the intended version of a software. If a new version is failing, try insalling an older version. 
+* __Building with an unexpected version of a dependency:__ If there are issues installing a dependency, you can modify the specs of a dependency.
+
+Additionally, spack will oftentimes output troubleshooting suggestions. To increase the verbosity of `spack install`, instead use `spack -dv install`. 
+
```

Comment:
Maybe include 'fill out our help form if you have any questions (with link to help form)'